src/controllers/cluster.rs

```diff
@@ -203,6 +218,27 @@ impl FleetBundle for FleetClusterBundle {
             Api::<BundleNamespaceMapping>::namespaced(ctx.client.clone(), &ns.unwrap_or_default())
                 .delete(&mapping.name_any(), &DeleteParams::default())
                 .await?;
+
+            // List all other clusters in this namespace
+            let namespaces = Api::<Namespace>::all(ctx.client.clone());
+            let namespace_name = mapping.namespace().unwrap_or_default();
+            let clusters = Api::<Cluster>::namespaced(ctx.client.clone(), &namespace_name);
+            let other_clusters = clusters
+                .list(&ListParams::default().fields(&format!("metadata.namespace={},metadata.name!={}", namespace_name, self.fleet.name_any())))
+                .await?;
```

Comment:
```suggestion
```
I think `other_clusters` can be cloned and reused, to avoid additional list request, as you do down below. That may require moving this outside of `if let Some(mapping)` block, otherwise this logic will only be executed on a cluster referencing a cluster-class from another namespace, which I don’t think is the goal here.

Reply:
@Danil-Grigorev doesn't`other_clusters` use the namespace of the `BundleNamespaceMapping` as opposed to the namespace of the cluster? https://github.com/rancher/cluster-api-addon-provider-fleet/blob/5aaff1b8b9b9ee57faa436401377470c5ef04088/src/controllers/cluster.rs#L184-L191